backend/appendlimit.go

```diff
@@ -0,0 +1,25 @@
+package backend
+
+import (
+	"errors"
+)
+
+// An error that should be returned by User.CreateMessage when the message size
+// is too big.
+var ErrTooBig = errors.New("Message size exceeding limit")
+
+// A backend that supports retrieving per-user message size limits.
+type AppendLimitBack interface {
```

Comment:
Should we embed the `Backend` interface here?

(Same goes for `User`)

Reply:
Most external extensions currently do not embed it. Though embedding it could be more convenient in some cases.  => We should

Reply:
https://github.com/emersion/go-imap/pull/342/commits/430faf9d88b20f3609b4aca3830f506bc98ba59b